src/Std/Internal/Async/Context.lean

```diff
@@ -0,0 +1,233 @@
+/-
+Copyright (c) 2025 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sofia Rodrigues
+-/
+module
+
+prelude
+public import Std.Time
+public import Std.Internal.UV
+public import Std.Internal.Async.Basic
+public import Std.Internal.Async.Timer
+public import Std.Sync.Context
+
+public section
+
+/-!
+This module contains the implementation of `ContextAsync`, a monad for asynchronous computations with
+built-in cancellation support.
+-/
+
+namespace Std
+namespace Internal
+namespace IO
+namespace Async
+
+/--
+An asynchronous computation with cancellation support via a `Context`. `ContextAsync α` is equivalent
```

Comment:
Should this be a bit more explicit that we are doing cooperative multitasking here, i.e., that computations need to check for cancellation themselves?